### PR TITLE
Goa 2241 regex for rna not working

### PR DESCRIPTION
**Pull request by TonyWardell**
I've added a test for case sensitivity for RNA, Intact and UniProt gene product ids. When loading the validation regexes in from file they work fine, which is why the jira request for the branch is closed, but this test proves this.

### CODE REVIEW

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/controller/AnnotationControllerIT.java [hunk 1]
```diff
@@ -400,7 +400,32 @@ public void filterByUniProtKBAndIntactAndRNACentralGeneProductIDSuccessfully() t
         response.andExpect(status().isOk())
                 .andExpect(contentTypeToBeJson())
                 .andExpect(totalNumOfResults(NUMBER_OF_GENERIC_DOCS))
-                .andExpect(fieldsInAllResultsExist(NUMBER_OF_GENERIC_DOCS));
+                .andExpect(fieldsInAllResultsExist(NUMBER_OF_GENERIC_DOCS))
```
Comment (rantunesboreas) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/controller/AnnotationControllerIT.java]: NUMBER_OF_GENERIC_DOCS should not be used here. Reason being that this value represents the number of documents stored in the repository before the test starts.

It just happens to match the number of documents that are retrieved (3). I would suggest creating a value within the test method, e.g. expectedNumberOfDocs = 3

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/controller/AnnotationControllerIT.java [hunk 2]
```diff
@@ -400,7 +400,32 @@ public void filterByUniProtKBAndIntactAndRNACentralGeneProductIDSuccessfully() t
...
+        repository.save(AnnotationDocMocker.createAnnotationDoc(intactGp));
+        repository.save(AnnotationDocMocker.createAnnotationDoc(rnaGp));
+        StringJoiner sj = new StringJoiner(",");
+        sj.add(uniprotGp).add(intactGp).add(rnaGp);
+        ResultActions response = mockMvc.perform(
+                get(RESOURCE_URL + "/search").param(GP_PARAM, sj.toString()));
+
+        response.andExpect(status().isOk())
+                .andExpect(contentTypeToBeJson())
+                .andExpect(totalNumOfResults(NUMBER_OF_GENERIC_DOCS))
```
Comment (rantunesboreas) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/controller/AnnotationControllerIT.java]: Same comment as above.

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/controller/AnnotationControllerIT.java (hunk 1 above)
Comment (TonyWardell) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/controller/AnnotationControllerIT.java]: I've in-lined all values. For many tests the expectedNumberOfDocs  just happens to be one of a number of numeric values to be tested, and the validation method names make intent clear.

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/controller/AnnotationControllerIT.java (hunk 2 above)
Comment (TonyWardell) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/controller/AnnotationControllerIT.java]: As above.

Comment (TonyWardell) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/controller/AnnotationControllerIT.java]: So code is back as it was, apart from the two changes you've requested. Creating the required annotation with the select-by data only in each test method would not work without removing the tests for completeness of model data, so I think they should remain in.

**Pull request merged**